### PR TITLE
User profile

### PR DESCRIPTION
**Pull request by xsaurabhsuthar**
image
image
image
image
image
* Explore, Home feed, bookmarks, user profile with edit options and user profile added.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -29,27 +35,32 @@ function App() {
             where("uid", "==", uid)
           );
 
-          const userInformation = await getDocs(docRef);
-          // it'll only have one doc
-          userInformation.forEach((doc) => {
-            console.log(doc.data());
-            dispatch(setUserInformation({ ...doc.data() }));
+          unsubscribeListner = onSnapshot(docRef, (docs) => {
+            console.log("data from snapshot");
```
Comment (saketkum11) [inline src/App.jsx:39]: if console would be remove it will make code much cleaner

File: src/common/index.js [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
 export { IconButton } from "./atoms/Button/IconButton/IconButton";
 export { InputField } from "./atoms/InputField/InputField";
+export { Loader } from "./atoms/Loader/Loader";
+export { OptionItem } from "./atoms/OptionItem/OptionItem";
 export { OutlineButton } from "./atoms/Button/OutlineButton/OutlineButton";
+export { SideBarNavLink } from "./atoms/SideBarNavLink/SideBarNavLink";
 export { SolidButton } from "./atoms/Button/SolidButton/SolidButton";
+export { Tab } from "./atoms/Tab/Tab";
+export { TextBoxField } from "./atoms/TextBoxField/TextBoxField";
+
```
Comment (saketkum11) [inline src/common/index.js:14]: great way to use barrel method

File: src/common/molecules/TextBox/TextBox.jsx [hunk 3]
```diff
@@ -79,15 +77,15 @@ export const TextBox = ({
...
-        dispatch(
-          updatePost({
-            post: {
-              ...updatedPost,
-              createdAt: updatedPost.createdAt.toDate().toString(),
-            },
-            _id: _id,
-          })
-        );
+        // dispatch(
```
Comment (saketkum11) [inline src/common/molecules/TextBox/TextBox.jsx:80]: you can remove this if not needed

File: src/hooks/usePosts.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+          docs.forEach((doc) => {
+            fetchedPosts.push({
+              _id: doc.id,
+              ...doc.data(),
+              createdAt: doc.data().createdAt.toDate().toString(),
+            });
+          });
+          updateHandler(fetchedPosts);
+
+          // setLastVisited(docs.data()[docs.data().length - 1]);
```
Comment (saketkum11) [inline src/hooks/usePosts.js:85]: you could remove this

**Review by saketkum11 — APPROVED** [review]
- Great code
- one suggestion could you keep PR code smaller because it deviate the focus of review code purly.

File: src/routes/Explore/ExploreSlice.js [hunk 5]
```diff
@@ -1,33 +1,63 @@
...
+// });
+
+// export const {
+//   explorePostsSortBy,
+//   setExplorePosts,
+//   appendExplorePosts,
+//   startLoadingExplorePosts,
+//   stopLoadingExplorePosts,
+// } = exploreSlice.actions;
+
```
Comment (manismk) [inline src/routes/Explore/ExploreSlice.js:36]: remove the commented if not needed

**Review by manismk — APPROVED** [review]
Looks good. Great usage of firebase and methods

**Pull request merged**